### ISSUE TITLE
Request to add support for tracking numbers of shipments

### ISSUE THREAD

**Issue by frmeier** [puhep/pudb#18]
We would like to have a field added to enter the tracking number when we ship out modules. It would be nice to have if this could be presented as a link to jump to the tracking webpage or even use their API to poll the delivery status.
We are using FedEx for our shipments. If other sites use a different provider, another field would be required to keep track of this.

**Comment by khan62** [comment]
I spent a good amount of time trying to integrate the FedEx API into the website. Unfortunately, we are missing a required php extension. I'm currently talking to the Purdue Physics IT department trying to get that extension installed, but barring that, I'll add a link to the tracking webpage.

**Comment by khan62** [comment]
While I wait on the IT guys, I've added a tracking number field to the "shipped" step. When that field is filled, a hyperlink called "Track Package" will appear on the part summary page, and it will send you to the FedEx tracking page.

**Comment by frmeier** [comment]
Did not yet try it (will ship out something later today) but I guess this is more than sufficient. Others may comment if they don't agree. More important is that we don't forget to populate it, and just being in a comment doesn't remind the user to fill in the tracking number.

**Comment by khan62** [comment]
The tracking number is a required field, so the module can't be promoted unless a tracking number is entered

**Comment by frmeier** [comment]
That is good, thanks. We also have the case where we actually bring modules to KU, where we won't have a tracking number. I guess we can enter a text 'courier', or do you check for a specific format?

**Comment by khan62** [comment]
It doesn't check format; it's a simple text field. I can set it up such that it won't generate a link if you enter "courier".

**Comment by frmeier** [comment]
Hmm. Maybe then it starts getting too fancy for that. Likely better the other way round. If a RegEx match reveals a properly formatted tracking number, generate the link. This might also allow to handle the difference if somebody uses a different provider.

**Comment by khan62** [comment]
PHP is actually pretty liberal about data types. It only took me an extra 2 lines to add the "Courier" functionality. I've also made it so that the link won't be displayed if the "tracking number" is non-numeric. It will be possible to add support for other carriers as well, assuming tracking numbers between carriers are distinct.
